src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -579,28 +583,30 @@ private String key() {
     private void finish(@Nonnull Result r, @CheckForNull Throwable t) {
         try {
             setResult(r);
+            BuildListener myListener;
             synchronized (getMetadataGuard()) {
+                myListener = getListener();
```

Comment:
These changes are not strictly necessary, but since I was looking at all uses of `getListener` anyway, I think it's much clearer to call `getListener` once inside of the synchronized block and then use the local variable rather than having this method call `getListener()` repeatedly in addition to accessing the `listener` field directly.